backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/dataimporter/ESRImportEnqueuer.java

```diff
@@ -174,9 +198,79 @@ public void execute()
 
 	}
 
-	private void checkUpdateDataType(final I_ESR_Import esrImport, final String fileName)
+	private void createImportFileFromSingleAttachment(@NonNull final I_ESR_Import esrImport,
+			final byte[] data,
+			@NonNull final String filename)
 	{
-		if (Check.isEmpty(esrImport.getDataType()))
+		final String hash = computeESRHashAndCheckForDuplicates(OrgId.ofRepoId(esrImport.getAD_Org_ID()), data);
+
+		final I_ESR_ImportFile esrImportFile = esrImportDAO.createESRImportFile(esrImport);
+		checkUpdateDataType(esrImportFile, filename);
+		esrImportFile.setHash(hash);
+
+		final AttachmentEntry attachmentEntry = attachmentEntryService.createNewAttachment(
+				esrImportFile,
+				filename,
+				data);
+		final AttachmentEntryId attachmentEntryId = attachmentEntry.getId();
+
+		esrImportFile.setAD_AttachmentEntry_ID(attachmentEntryId.getRepoId());
+		esrImportFile.setFileName(filename);
+		esrImportDAO.save(esrImportFile);
+	}
+
+	private void createImportFilesFromZips(@NonNull final I_ESR_Import esrImport,
+			@NonNull final ByteArrayInputStream in,
+			@NonNull final String filename)
+	{
+		final ZipInputStream zipStream = new ZipInputStream(in);
+
+		try
+		{
+			ZipEntry zipEntry = zipStream.getNextEntry();
+			while (zipEntry != null)
+			{
+				if (zipEntry.isDirectory())
+				{
+					zipEntry = zipStream.getNextEntry();
+					continue;
+				}
+				final Resource unzippedFile = extractResource(zipStream, zipEntry);
```

Comment:
u could change the return type as ZipFileResource.
and introduce `byte[] ZipFileResource.getData()` which u could use it couple of lines below.